index.js

```diff
@@ -1,10 +1,332 @@
 module.exports = (robot) => {
-  // Your code here
-  robot.log('Yay, the app was loaded!')
-
-  // For more information on building apps:
-  // https://probot.github.io/docs/
-
-  // To get your app running against GitHub, see:
-  // https://probot.github.io/docs/development/
-}
+  robot.log('Yay, the app was loaded!');
+  robot.on('*', async context => {
+    const event = context.event;
+    const action = context.payload.action;
+    robot.log("Event: " + event + ((typeof action === 'undefined' ||  action === null)? "" : (", Action: " + action)));
+    switch(event){

```

Comment:
I thought that,  when any event occurs, maybe, this will trigger a webhook and I will be storing the useful data in `filename-week1.yml` on weekly basis.
My idea: 
Keep storing data in suitable formats in `filename-week1.yml` on weekly basis initially, and at the end of the week, go through this file and generate a weekly report on this basis.
After generating the weekly report, the app creates a filename-week2.yml` and and saves next events there?
This is my thoughts, and I did the above code for this idea only, what are your thoughts?

Reply:
A doubt.
> I think a simpler approach would be to not listen to any events but setup a (configurable) schedule which will call the method that goes trough all the data that you might want to use for the digest. That way you don’t need your own database

So, shouldn't I store the weekly digest data in some file?

Reply:
Your idea with storing the data in a file until it’s used is quite clever, it will make it easier to compare differences from week to week. 

One thing to be aware of though is that you might hit rate limiting pretty quickly. Updating a file is a write request, you shouldn’t send more than one per second, and if you do write once per second consistantly you might hit other rate limts and be blocked by GitHub. So you could wait a few seconds at a time, store the data in memory and then write the file once every 30seconds or so.

I’d suggest to implement the pulling first, it will be simpler to do. Storing the data can be something for a later iteration. 

What do you think?

Reply:
Yeah, @gr2m , I missed the fact that this might hit data rate limit. And it's really important point that should be considered. 🙂 

For initial phase, I will try implementing  something like [pulse](https://github.com/probot/probot/pulse) and maybe then later move to the next step.

Reply:
sounds good :)